doc_src/cmds/bind.rst

```diff
@@ -362,21 +379,7 @@ Turn on :ref:`vi key bindings <vi-mode>` and rebind :kbd:`Control`\ +\ :kbd:`C`
 
 Launch ``git diff`` and repaint the commandline afterwards when :kbd:`Control`\ +\ :kbd:`G` is pressed::
 
-   bind \cg 'git diff' repaint
-
-.. _cmd-bind-termlimits:
-
-Terminal Limitations
---------------------
-
-Unix terminals, like the ones fish operates in, are at heart 70s technology. They have some limitations that applications running inside them can't workaround.
```

Comment:
This would need something on CSI u and then *still* something on the limitation of the legacy system.

I do not believe that CSI u is that well-supported that we can ignore the older system almost entirely.

Reply:
I guess you're right but I don't want to spend too much time on this.
Given that, I might not even mention CSI u - it's an implementation detail and it should just work, the same as in Vim and elsewhere.
So I'd prefer to let someone else write the documentation if they feel like it's useful.
I guess I can avoid touching the existing docs.
I plan to switch to a different flavor of the protocol in future.

Reply:
How about something like:

```
fish works best in terminal emulators that support the modern CSI u key sequence protocol.

It will try to enable this automatically, and if your terminal supports it everything should just work.

If your terminal does not support CSI u, key bindings will have some limitations:

- being incapable of telling many characters + control from other keys, like Control+I from tab or Control+J from newline.
- Control and Shift can't be used simultaneously. 
- not knowing some modifier+key sequences

These limitations are caused by the old encoding simply not giving the application more information, which is why the CSI u protocol was created.
```

Reply:
Yeah those are good things to mention.
It's not exactly CSI u but a mix of two protocols with kitty extensions. It's subject to change in future, so I'm not sure if I want to document the specifics. But I think linking to the kitty page is a good idea.

Reply:
I don't really want to link to kitty documentation because that makes it look like a single-terminal thing.

> It's not exactly CSI u but a mix of two protocols with kitty extensions.

Okay, so which things do we need, what terminals does this work properly on? What can we mention here?

Reply:
Features like disambiguation of keys `ctrl-j`  or `shift-enter` work only on terminals that implement [kitty extensions](https://sw.kovidgoyal.net/kitty/keyboard-protocol/).

(But users of all terminals benefit from having a human-friendly, (terminal/protocol)-agnostic key binding syntax)

> I don't really want to link to kitty documentation because that makes it look like a single-terminal thing.

I think this is very easy to avoid, like "if your terminal implements the kitty keyboard protocol ...".
I don't think it's very important to document the specifics because it's always easy to get the ground truth by trying fish on some terminal.
It should be obvious that "if the terminal disambiguates a key, then you can map it" because I don't think there is any other cross-terminal protocol around (not counting terminfo).
When a new protocol comes along we can implement that etc.

Reply:
XTerm users also get some benefits from `modifyOtherKeys`. I dropped this doc change for now

Reply:
> It should be obvious that "if the terminal disambiguates a key, then you can map it" because I don't think there is any other cross-terminal protocol around (not counting terminfo).

What I'm looking for here is some feature we can tell people to look for in a terminal without having to try it out and see.

Ideally this would just be a bullet point like "Keyboard Input Protocol 2024". I don't love calling it the "kitty keyboard protocol" because that pushes people towards kitty specifically, but if that's the best description we have I guess that's it.

> I dropped this doc change for now

That's fine.